src/pages/index.astro

```diff
@@ -5,6 +5,9 @@ import SideMenu from '../components/SideMenu.astro';
 
 import { actions } from 'astro:actions';
 const time = await Astro.callAction(actions.serverTime, undefined);
+
+import { app } from '../lib/firebase.ts';
+console.log("Firebase App:", app); // debug use
```

Comment:
The CI issue comes from you trying to spawn and use Firebase at server render time.